File: projects/10/java/Penguin.java
/* Penguin.java provides a Goose subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Daniela Contreras
 * Date: 03/30/22
 ******************************************************/

public class Penguin extends WalkingBird {

    /* explicit constructor
     * Receive: name, a String 
     * PostCond: myName == name.
     */
     public Penguin()
     {
        super();
     }
   
     public Penguin(String name)
     {     
       super(name);
     } 
   
    /* A Penguin's Call
     * Return: a goose-call ("Huh-huh-huh-huuuuh!").
     */
   
     public String call()
     {
       return "Huh-huh-huh-huuuuh!";
     } 
   
   
   }

File: projects/10/ruby/Penguin.rb
# Penguin.rb | Defines the Peguin class which inherits attributes and methods
#   from the WalkingBird superclass.
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Daniela Contreras
# Date: 03/29/22
####################################################

require './WalkingBird.rb'

class Penguin < WalkingBird

##############################################
## Call returns sound of Duck                 #
##Return: Quack! a string                    #
##############################################
    def call
        'huh-huh-huh-huh!'
    end
    
end